### ISSUE TITLE
Facebook login with no e-mail

### ISSUE THREAD

**Issue by mferak** [CakeDC/users#430]
Hi.
My app is set up in a way where I'm able to register & login using the Users plugin - either via classical account OR via a Facebook account. Thus, everything should be set up properly.
However, when I try to register&login with a Facebook account that has no associated e-mail address, I end up in the `socialEmail` method in `SocialTrait`. This fails on line 32 - the session is empty, throwing a `NotFoundException`, not giving me the option to enter custom e-mail. I've tried to debug using xdebug, found the place where the session gets destroyed: it happens in `SocialAuthenticate`'s method `getUser` on line 397. Not sure why though.

For reference, here's my custom users.php config:

`'Users.Social.login' => true,
  'Users.table' => 'Users',
  'Users.Email.validate' => true,
  'Users.Profile.route' => ['controller' => 'Users', 'action' => 'edit'],
  'Users.reCaptcha.key' => '...',
  'Users.reCaptcha.secret' => '...',
  'Users.reCaptcha.registration' => false,
  'Users.Tos.required' => true,
  'OAuth.providers.facebook.options.clientId' => '...',
  'OAuth.providers.facebook.options.clientSecret' => '...',
  'Auth.loginRedirect' => '/',
  'Auth.logoutRedirect' => '/',`

Has anyone else experienced this issue?

Thanks for any feedback.

**Comment by steinkel** [comment]
Facebook login assumes now that you have a valid email address available, so the email request workflow is not triggered. We would need to modify the login process to allow facebook users without email to be able to register.

**Comment by mferak** [comment]
Funny thing I noticed: we have a client whose email address shows up on facebook as valid and verified, yet this problem persists. When we added a new email to his account, and verified that one, we were able to log in correctly. Seems like the old email they used on their account was no longer active, and Facebook might have internally marked it down as bounced - which resulted in it no longer being retrieved by the oauth api - this however was not indicated at all in the user's FB settings.
